docs/technical-reference/tasks.md

```diff
@@ -0,0 +1,159 @@
+
+Contrary to the `TextGenerationTask`, the `PreferenceTask` is not intended for direct use. It implements the default methods `input_args_names` and `output_args_names`, but `generate_prompt` and `parse_output` are specific to each `PreferenceTask`. Examining the output_args_names reveals that the generation will encompass both the rating and the rationale that influenced that rating. `distilabel` implements the following preference tasks:
+
+#### UltraFeedbackTask
+
+The task specially designed to build the prompts following the UltraFeedback paper: [UltraFeedback: Boosting Language Models With High Quality Feedback](https://arxiv.org/pdf/2310.01377.pdf).
+
+From the original [repository](https://github.com/OpenBMB/UltraFeedback): *To collect high-quality preference and textual feedback, we design a fine-grained annotation instruction, which contains 4 different aspects, namely instruction-following, truthfulness, honesty and helpfulness*. This `Task` is designed to label datasets following the different aspects defined for the UltraFeedback dataset creation.
+
+The following snippet can be used as a simplified UltraFeedback Task, for which we define 3 different ratings, but take into account the predefined versions are intended to be used out of the box:
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/ultrafeedback.py"
+```
+
+- Text Quality:
+
+The following example uses a `LLM` to examinate the data for text quality criteria, which includes the different criteria from UltraFeedback (Correctness & Informativeness, Honesty & Uncertainty, Truthfulness & Hallucination and Instruction Following):
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/openai_for_text_quality.py"
+```
+
+- Helpfulness:
+
+The following example creates a UltraFeedback task to emphasize helpfulness, that is overall quality and correctness of the output:
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/openai_for_helpfulness.py"
+```
+
+- Truthfulness:
+
+The following example creates a UltraFeedback task to emphasize truthfulness and hallucination assessment:
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/openai_for_truthfulness.py"
+```
+
+- Honesty:
+
+The following example creates a UltraFeedback task to emphasize honesty and uncertainty expression assessment:
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/openai_for_honesty.py"
+```
+
+- Instruction Following:
+
+The following example creates a UltraFeedback task to emphasize the evaluation of alignment between output and intent:
+
+```python
+--8<-- "docs/snippets/technical-reference/tasks/openai_for_instruction_following.py"
+```
+
+For the API reference visit [UltraFeedbackTask][distilabel.tasks.preference.ultrafeedback.UltraFeedbackTask].
```

Comment:
Maybe we can keep these collapsed using content tabs? See more at https://squidfunk.github.io/mkdocs-material/reference/content-tabs/#linked-content-tabs

Reply:
yeah:
![image](https://github.com/argilla-io/distilabel/assets/56895847/a74394d5-98e4-42dc-bcd3-c7f5852623f3)